core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,628 @@ def update_question_suggestion(
+    contribution stats.
+
+    Args:
+        question_contribution_stats: list(QuestionContribution). A list of
+        QuestionContribution domain objects.
+    """
+    stats_ids = []
+
+    for stat in question_contribution_stats:
+        stats_ids.append(
+            suggestion_models.QuestionContributionStatsModel.generate_id(
+                stat.contributor_user_id,
+                stat.topic_id
+            ))
+
+    stats_models_to_update = (
+        suggestion_models.QuestionContributionStatsModel.get_multi(stats_ids))
+
+    for index, stats_model in enumerate(stats_models_to_update):
+        stat = question_contribution_stats[index]
+        stats_model.submitted_questions_count = (
+            stat.submitted_questions_count)
+        stats_model.accepted_questions_count = (
+            stat.accepted_questions_count)
+        stats_model.accepted_questions_without_reviewer_edits_count = (
+            stat.accepted_questions_without_reviewer_edits_count)
+        stats_model.first_contribution_date = stat.first_contribution_date
+        stats_model.last_contribution_date = stat.last_contribution_date
+
+    suggestion_models.QuestionContributionStatsModel.update_timestamps_multi(
+        stats_models_to_update,
+        update_last_updated_time=True)
+    suggestion_models.QuestionContributionStatsModel.put_multi(
+        stats_models_to_update)
+
+
+def _update_question_review_stats_models(question_review_stats):
+    """Updates QuestionReviewStatsModel models for given question
+    review stats.
+
+    Args:
+        question_review_stats: list(QuestionReviewStats). A list of
+        QuestionReviewStats domain objects.
+    """
+    stats_ids = []
+
+    for stat in question_review_stats:
+        stats_ids.append(
+            suggestion_models.QuestionReviewStatsModel.generate_id(
+                stat.contributor_user_id,
+                stat.topic_id
+            ))
+
+    stats_models_to_update = (
+        suggestion_models.QuestionReviewStatsModel.get_multi(stats_ids))
+
+    for index, stats_model in enumerate(stats_models_to_update):
+        stat = question_review_stats[index]
+        stats_model.reviewed_questions_count = (
+            stat.reviewed_questions_count)
+        stats_model.accepted_questions_count = (
+            stat.accepted_questions_count)
+        stats_model.accepted_questions_with_reviewer_edits_count = (
+            stat.accepted_questions_with_reviewer_edits_count)
+        stats_model.first_contribution_date = stat.first_contribution_date
+        stats_model.last_contribution_date = stat.last_contribution_date
+
+    suggestion_models.QuestionReviewStatsModel.update_timestamps_multi(
+        stats_models_to_update,
+        update_last_updated_time=True)
+    suggestion_models.QuestionReviewStatsModel.put_multi(
+        stats_models_to_update)
+
+
+def update_translation_contribution_stats(
+    suggestion, user_id, target_id, is_submission):
```

Comment:
The `is_submission` adds an unnecessary complexity IMO. Can we have a separate function for submission and not submission?

Reply:
Done